libs/blocks/adobetv/adobetv.js

```diff
@@ -7,13 +7,21 @@ const loadAdobeTv = (a) => {
   const bgBlocks = ['aside', 'marquee', 'hero-marquee'];
   if (a.href.includes('.mp4') && bgBlocks.some((b) => a.closest(`.${b}`))) {
     a.classList.add('hide');
+    const parentElement = a.parentNode;
+    if (!parentElement) return;
     const { href, hash, dataset } = a;
     const attrs = getVideoAttrs(hash || 'autoplay', dataset);
-    const video = `<video ${attrs}>
-          <source src="${href}" type="video/mp4" />
-        </video>`;
-    if (!a.parentNode) return;
+    const video = `<video ${attrs}></video>`;
     a.insertAdjacentHTML('afterend', video);
+    createIntersectionObserver({
```

Comment:
> Is there any way we could use decorate.js to avoid repetition?

@overmyheadandbody While we could move it into decorate, it's used by more than just the video blocks. IMO it already contains methods it should not and it's "growing" instead of `utils.js` ...

I don't want to shift the utils.js problem into another file, so IMO the better call is to live with the duplication. Both of the blocks seem similar enough to me, so that they could be the same thing 🤷 But that's too late to change

Reply:
If the video files are the only ones using then, would it be then worth having a video utils file that has the common methods and then take them out of `decorate.js`? I agree that we don't want to needlessly bloat common files, but I also think that having to duplicate functionality is not maintainable. If your tests show that loading yet another video-utils.js file or whatever comes with some performance penalties, then we can just go with duplication, but I'd like to get some insights firts.

Reply:
I couldn't really split the files, however after testing things around, it's a lot cleaner moving the logic in decorate. Those differences are hard to maintain and already lead to a bug in the past by forgetting to set sth up correctly.